Upgrade to 1.9.0 version of the native library

VladimirAmiorkov opened the pull request:
<!--
We, the rest of the NativeScript community, thank you for your
contribution! 
To help the rest of the community review your change, please follow the instructions in the template.
-->

<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

## PR Checklist

- [X] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [X] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [X] All existing tests are passing
- [ ] Tests for the changes are included

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
The current version uses a pre-release version 0.9.0+ of the native Android Fresco library. Also sometimes setting the `imageUri` to an incorrect value causes exception that is not descriptive enough.

## What is the new behavior?
<!-- Describe the changes. -->
The plugin now uses the latest official 1.9.0 version of the native Android Fresco library as documented [here](http://frescolib.org/docs/index.html).

Fixes/Implements/Closes #[Issue Number].
#59
#16 

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## BREAKING CHANGES:

- `AnimatedImage` class has been removed
- `IAnimatedImage`  interface has been renamed to `AnimatedImage` 
- `IImageInfo` interface has been renamed to `ImageInfo` 
- `IError` interface has been renamed to `FrescoError` 

### Migration
- If you are using the `AnimatedImage` class you can continue to cast to this type just note that it is now an interface rather than a class
- If you are using the `IAnimatedImage` interface simply change all of your references to be `AnimatedImage`
- If you are using the `IImageInfo` interface simply change all of your references to be `ImageInfo`
- If you are using the `IError` interface simply change all of your references to be `FrescoError`

VladimirAmiorkov committed:
chore: Upgrade native library to 1.9.0

VladimirAmiorkov committed:
chore: Upgrade demo with import for native Fresco typings
chore: Add npmignore
fix: Add warrning message if imageUri is set to incorrect value
chore: Update tsconfig file according to plugin seed

VladimirAmiorkov committed:
chore: Remove commented out code

VladimirAmiorkov committed:
chore: Bump version

lini commented on src/platforms/android/include.gradle:
```diff
@@ -1,11 +1,8 @@
 android {
-	productFlavors {
```
productFlavors are needed for backwards compatibility with {N} 3.x

VladimirAmiorkov replied:
Yes of course, this was an auto change by the tns run command

lini commented on src/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "nativescript-fresco",
-  "version": "3.1.0",
+  "version": "4.0.0",
```
Major version is expected to have breaking changes. If it does, they should be described in a migration guide in the plugin's readme (see imagepicker for example)

VladimirAmiorkov replied:
Added breaking changes to PR and readme

VladimirAmiorkov committed:
chore: Returned productFlavors

VladimirAmiorkov committed:
chore: Returned the 'IAnimatedImage' to be used rather than native class in wrapper

VladimirAmiorkov committed:
docs: Update readme with migration guide from 3.x.x to 4.x.x

VladimirAmiorkov committed:
chore: Add example that shows how to manually start animating image

VladimirAmiorkov committed:
chore: Rename interfaces to TypeScript best practices
docs: Update readme with migration process
chore: Made demo logs more informative

VladimirAmiorkov committed:
chore: Rename interface name in d.ts
